### ISSUE TITLE
Clarify policy on breaking changes

### ISSUE THREAD

**Issue by chaosagent** [nix-rust/nix#458]
Theres no general policy on breaking changes in CONTRIBUTING.md; I think it'd be useful to have one.

**Comment by fiveop** [comment]
I think the current consensus is, that breaking things in order to make the code follow CONVENTIONS.md or to fix some problem (not necessarily a bug, see #445 for example) is fine. When we start to think about a 1.0 version, that will change. Feel free to write a paragraph stating that as a PR, that we can then further discus. If you don't, I might get around to it next week.

**Comment by posborne** [comment]
Yep, at this point we don't want to break APIs unless there is a good reason.  But if there is a good reason, then I think it still makes sense to do so (and semver will change accordingly).

**Comment by kamalmarhubi** [comment]
Thanks for asking about this. We should definitely figure this out.
